File: packages/urdu-js/src/string-utils/count.ts
import { stripDiacritics } from '../normalization/diacritics.js'

export function wordCount(text: string): number {
  return text.trim() ? text.trim().split(/\s+/).length : 0
}

export function charCount(
  text: string,
  options?: { excludeDiacritics?: boolean; excludeWhitespace?: boolean },
): number {
  let s = text
  if (options?.excludeDiacritics) s = stripDiacritics(s)
  if (options?.excludeWhitespace) s = s.replace(/\s/g, '')
  return [...s].length
}

File: packages/urdu-dart/lib/src/string_utils/count.dart
/// Returns the number of words in the text, splitting on whitespace.
int wordCount(String text) {
  if (text.trim().isEmpty) return 0;
  return text.split(RegExp(r'\s+')).where((p) => p.isNotEmpty).length;
}

/// Returns the exact number of characters (not bytes).
int charCount(String text) => text.runes.length;
